docs/framework-integrations/nextjs.mdx

```diff
@@ -0,0 +1,424 @@
+---
+slug: /nextjs
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Next.js
+
+Integration guide for the [React][] components for the Uppy UI plugins and
+hooks.
+
+:::tip
+
+Uppy also has hooks and more React examples in the [React docs](/docs/react).
+
+:::
+
+## Install
+
+<Tabs>
+  <TabItem value="npm" label="NPM" default>
+
+```shell
+npm install @uppy/core @uppy/dashboard @uppy/react
+```
+
+  </TabItem>
+
+  <TabItem value="yarn" label="Yarn">
+
+```shell
+yarn add @uppy/core @uppy/dashboard @uppy/react
+```
+
+  </TabItem>
+</Tabs>
+
+## Tus
+
+[Tus][tus] is an open protocol for resumable uploads built on HTTP. This means
+accidentally closing your tab or losing connection let’s you continue, for
+instance, your 10GB upload instead of starting all over.
+
+Tus supports any language, any platform, and any network. It requires a client
+and server integration to work. We will be using [tus Node.js][].
+
+Checkout the [`@uppy/tus` docs](/docs/tus) for more information.
+
+```tsx
+'use client';
+
+import Uppy from '@uppy/core';
+// For now, if you do not want to install UI components you
+// are not using import from lib directly.
+import Dashboard from '@uppy/react/lib/Dashboard';
+import Tus from '@uppy/tus';
+import { useState } from 'react';
+
+import '@uppy/core/dist/style.min.css';
+import '@uppy/dashboard/dist/style.min.css';
+
+function createUppy() {
+	return new Uppy().use(Tus, { endpoint: '/api/upload' });
```

Comment:
Looks like we're using tabs for indentation, if we switch to tabs it should be across the board i think, not just this doc.

Reply:
This is all auto formatted so should be inline with everything else.